File: go/vertical_slide/features/register_account/repository.go
package register_account

import (
	"ex/vertical_slide/shared/model"
	"ex/vertical_slide/shared/repository"

	"github.com/scylladb/gocqlx/v2/qb"
	scyna "github.com/scyna/core"
)

type Repository struct {
	repository.BaseRepository
}

func NewRepository(LOG scyna.Logger) *Repository {
	return &Repository{
		BaseRepository: *repository.NewBaseRepository(LOG),
	}
}

func (r *Repository) CreateAccount(account *model.Account) scyna.Error {

	if err := qb.Insert(repository.ACCOUNT_TABLE).
		Columns("id", "name", "email", "password").
		Query(scyna.DB).
		BindStruct(account).
		ExecRelease(); err != nil {
		r.LOG.Error(err.Error())
		return scyna.SERVER_ERROR
	}
	return nil
}

File: csharp/vertical_slide/features/RegisterAccount/Repository.cs
namespace ex.vsa.CreateAccount;

using scyna;
using ex.vsa.model;
using ex.vsa.repository;

public class Repository : BaseRepository
{
    private Repository(Context context) : base(context) { }
    public static Repository create(Context context) { return new Repository(context); }

    public void AssureAccountNotExists(string email)
    {
        try
        {
            this.GetAccountByEmail(email);
        }
        catch (scyna.Error e)
        {
            if (e != model.Error.ACCOUNT_NOT_FOUND) throw;
        }
    }

    public void RegisterAccount(Account account)
    {
        var query = Engine.DB.Session.Prepare(string.Format("INSERT INTO {0}.{1}(id,email,name,password) VALUES(?,?,?,?)", KEYSPACE, ACCOUNT_TABLE));
        var statement = query.Bind((long)account.ID, account.Email, account.Name, account.Password);
        Engine.DB.Session.Execute(statement);
    }
}